### PR TITLE
[IMP] mail: convert timer utils into model

### PR DESCRIPTION
**Pull request by alexkuhn**
Better integration with the rest of the code.

Task-2831082

### CODE REVIEW

File: addons/mail/static/src/models/timer.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+        'threadAsCurrentPartnerInactiveTypingInThreadTimerOwner',
+        'threadAsCurrentPartnerLongTypingInThreadTimerOwner',
+    ]],
+    lifecycleHooks: {
+        _willDelete() {
+            this.messaging.browser.clearTimeout(this.timeoutId);
+            if (!this.timeoutDeferred) {
+                return;
+            }
+            this.timeoutDeferred.reject(new TimerClearedError());
```
Comment (alexkuhn) [inline addons/mail/static/src/models/timer.js:22]: Note: it was using `this.id` in previous code, but it was never introduced, so it was always `undefined`.
I just dropped this.

**Comment by seb-odoo** [comment]
I fixed runbot + cleaned code, we should review changes, squash, and decide if we want to do anything else (rename model?)

File: addons/mail/static/src/models/thread.js [hunk 2]
```diff
@@ -925,13 +926,7 @@ registerModel({
...
         registerOtherMemberTypingMember(partner) {
-            const timer = new Timer(
-                this.messaging,
-                () => this._onOtherMemberLongTypingTimeout(partner),
-                60 * 1000
-            );
-            this.otherMembersLongTypingTimers.set(partner, timer);
-            timer.start();
+            this.update({ otherMembersLongTypingTimers: insert({ partner: replace(partner) }) });
             const newOrderedTypingMemberLocalIds = this.orderedTypingMemberLocalIds
```
Comment (seb-odoo) [inline addons/mail/static/src/models/thread.js:930]: note: `orderedTypingMemberLocalIds`, `isTyping` and other such fields can probably be removed in favor of the existence of the timer(s)

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Review by seb-odoo — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**